tests/foreman/api/test_docker.py

```diff
@@ -638,44 +543,44 @@ def test_add_multiple_docker_repos_to_composite_content_view(self):
+        cv_versions = []
+        products = [
+            entities.Product(organization=self.org).create()
+            for _
+            in range(randint(1, 5))
+        ]
+        for product in products:
+            repo = _create_repository(product)
 
-        for _ in range(randint(1, 5)):
-            prod_ids.append(
-                entities.Product(organization=self.org_id).create_json()['id']
+            # Create content view and associate docker repo
+            content_view = entities.ContentView(
+                composite=False,
+                organization=self.org,
+            ).create()
+            content_view.repository = [repo]
+            content_view = content_view.update(['repository'])
+            self.assertIn(
+                repo.id,
+                [repo_.id for repo_ in content_view.repository]
             )
 
-        for prod_id in prod_ids:
-            repo_id = _create_repository(prod_id)['id']
+            # Publish it and grab its version ID (there should be one version)
+            content_view.publish()
+            content_view = content_view.read()
+            cv_versions.append(content_view.version[0])
 
-            # Create content view and associate docker repo
-            content_view = entities.ContentView(
-                organization=self.org_id, composite=False
-            ).create_json()
-            _add_repo_to_content_view(repo_id, content_view['id'])
-
-            new_attrs = entities.ContentView(id=content_view['id']).read_json()
-            self.assertIn(repo_id, new_attrs['repository_ids'])
-
-            # Publish it...
-            entities.ContentView(id=content_view['id']).publish()
-            # ... and grab its version ID (there should only be one version)
-            new_attrs = entities.ContentView(id=content_view['id']).read_json()
-            cv_version_ids.append(new_attrs['versions'][0]['id'])
-
-        # Create composite content view and associate content view to
-        # it
-        comp_content_view_id = entities.ContentView(
-            organization=self.org_id, composite=True
-        ).create_json()['id']
-        for version_id in cv_version_ids:
-            _add_content_view_to_composite_view(
-                comp_content_view_id, version_id)
-
-            new_attrs = entities.ContentView(
-                id=comp_content_view_id).read_json()
-            self.assertIn(version_id, new_attrs['component_ids'])
+        # Create composite content view and associate content view to it
+        comp_content_view = entities.ContentView(
+            composite=True,
+            organization=self.org,
+        ).create()
+        for cv_version in cv_versions:
+            comp_content_view.component.append(cv_version)
+            comp_content_view = comp_content_view.update(['component'])
+            self.assertIn(
+                cv_version.id,
+                [component.id for component in comp_content_view.component]
```

Comment:
The `update()` method returns a fresh content view object, just like `create()` and `read()` do.